worlds/witness/__init__.py

```diff
@@ -210,31 +195,64 @@ def create_regions(self) -> None:
         state = CollectionState(self.multiworld)
         state.sweep_for_events(locations=event_locations)
 
-        num_early_locs = sum(1 for loc in self.multiworld.get_reachable_locations(state, self.player) if loc.address)
+        # Adjust the needed size for sphere 1 based on how restrictive the settings are in terms of items
+
+        early_locations = [
+            location for location in self.multiworld.get_reachable_locations(state, self.player)
+            if not location.is_event
+        ]
+        self.reachable_early_locations = [location.name for location in early_locations]
+
+        num_reachable_locations = len(early_locations)
+        num_reachable_tutorial_locations = sum(loc.parent_region.name == "Tutorial" for loc in early_locations)
 
         # Adjust the needed size for sphere 1 based on how restrictive the settings are in terms of items
 
-        needed_size = 3
-        needed_size += self.options.puzzle_randomization == "sigma_expert"
-        needed_size += self.options.shuffle_symbols
-        needed_size += self.options.shuffle_doors > 0
+        needed_size_overall = 3
+        needed_size_overall += self.options.puzzle_randomization == "sigma_expert"
+        needed_size_overall += self.options.shuffle_symbols
+        needed_size_overall += self.options.shuffle_doors > 0
+
+        needed_size_to_hold_tutorial_items = len(self.options.early_good_items.value)
 
         # Then, add checks in order until the required amount of sphere 1 checks is met.
 
+        extra_tutorial_checks = [
+            ("Tutorial First Hallway Room", "Tutorial First Hallway Bend"),
+            ("Tutorial First Hallway", "Tutorial First Hallway Straight")
+        ]
+
         extra_checks = [
             ("Tutorial First Hallway Room", "Tutorial First Hallway Bend"),
             ("Tutorial First Hallway", "Tutorial First Hallway Straight"),
             ("Desert Outside", "Desert Surface 1"),
             ("Desert Outside", "Desert Surface 2"),
         ]
 
-        for i in range(num_early_locs, needed_size):
+        for i in range(num_reachable_tutorial_locations, needed_size_to_hold_tutorial_items):
+            if not extra_tutorial_checks:
+                break
+
+            region, loc = extra_tutorial_checks.pop(0)
+            extra_checks.pop(0)
+            self.player_locations.add_location_late(loc)
+            self.get_region(region).add_locations({loc: self.location_name_to_id[loc]}, WitnessLocation)
+            self.reachable_early_locations.append(loc)
+
+            player = self.multiworld.get_player_name(self.player)
```

Comment:
This should work, no?
```suggestion
            player = self.player_name
```